plugins/controlboard/src/ControlBoardDriverCoupling.cpp

```diff
@@ -564,3 +565,239 @@ yarp::sig::Vector CerHandCouplingHandler::decoupleRefTrq (yarp::sig::Vector& trq
     out[m_coupledJoints[3]] = trq_ref[m_coupledJoints[1]]/2;
     return out;
 }
+
+//------------------------------------------------------------------------------------------------------------------
+// HandMk3CouplingHandler
+//------------------------------------------------------------------------------------------------------------------
+
+HandMk3CouplingHandler::HandMk3CouplingHandler(gazebo::physics::Model* model, yarp::sig::VectorOf<int> coupled_joints, std::vector<std::string> coupled_joint_names)
+: BaseCouplingHandler(model, coupled_joints,coupled_joint_names), LUTSIZE(4096)
+{
+    const double RAD2DEG = 180.0/atan2(0.0,-1.0);
+    const double DEG2RAD = 1.0/RAD2DEG;
+    
+    m_couplingSize = 11;
+    
+    thumb_lut = new double[LUTSIZE];
+    index_lut = new double[LUTSIZE];
```

Comment:
Unfortunately manually managing this memory means that we would need to also define a set of custom copy constructors, copy operator, move constructors and move operators. I guess that perhaps it could be more simple to just use a `std::vector` and observe the [rule of zero](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rc-zero).

Reply:
@traversaro done.

Reply:
Thanks!